pkg/reconciler/knativeserving/common/images.go

```diff
@@ -150,6 +150,10 @@ func updatePodImagePullSecrets(deployment *appsv1.Deployment, registry *servingv
 	if len(registry.ImagePullSecrets) > 0 {
 		log.Debugf("Adding ImagePullSecrets: %v", registry.ImagePullSecrets)
 		existingSecrets := deployment.Spec.Template.Spec.ImagePullSecrets
-		deployment.Spec.Template.Spec.ImagePullSecrets = append(registry.ImagePullSecrets, existingSecrets...)
+		if len(existingSecrets) > 0 {
```

Comment:
I do not think you need to check the length of the existing secrets, you can directly use:

Spec.Template.Spec.ImagePullSecrets = append(
			Spec.Template.Spec.ImagePullSecrets, registry.ImagePullSecrets...)

Reply:
Thanks, updated!